custom/icds_reports/management/commands/phone_info_grouped_by_state_district.py

```diff
@@ -0,0 +1,150 @@
+import csv
+
+from django.core.management.base import BaseCommand
+from django.db import connections
+from datetime import datetime
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.utils.connections import get_icds_ucr_citus_db_alias
+
+"""
+The way query variables are structured is, they will have a fixed starting and a fixed end.
+And the middle condition is variable based on the type of beneficary we are targetting.
+"""
+
+COMMON_QUERY_START = """SELECT
+person_cases.state_id,
+person_cases.district_id,
+COUNT(*) "count",
+SUM(
+    CASE
+        WHEN person_cases.phone_number SIMILAR TO '[6789][0-9]{9}' THEN 1 ELSE 0
+    END) "valid_phone_count",
+SUM(
+    CASE WHEN (person_cases.phone_number != ''  AND person_cases.phone_number IS NOT NULL)  AND phone_number NOT SIMILAR to '[6789][0-9]{9}' THEN 1 ELSE 0
+    END) "invalid_phone_count",
+SUM(
+    CASE
+        WHEN (person_cases.phone_number = '') IS NOT FALSE  THEN 1 ELSE 0
+    END) "phone_not_available"
+ """
+
+COMMON_QUERY_END = """ and person_cases.migration_status=0 and person_cases.closed_on is null and person_cases.registered_status=1
```

Comment:
you should add `person_cases.closed_on is null` filter along with `person_cases.doc_id is not null` i.e
from `person_cases.closed_on is null` to `person_cases.doc_id is not null and person_cases.closed_on is null`. Because this person ucrs is going in the right of left join and by any chance for a particular ccs_case(for say) person case is missing from the ucr `person_cases.closed_on is null` would give you true but you don't want to include that in calculation if person case is not there